amti/actions/review.py

```diff
@@ -140,7 +171,15 @@ def review_batch(
 
     logger.info(f'Reviewing batch {batch_id}.')
 
+    marked_assignments = []
     for hit_id in hit_ids:
-        review_hit(client=client, hit_id=hit_id, approve_all=approve_all)
+        marked_assignments.extend(review_hit(client=client, hit_id=hit_id, approve_all=approve_all))
+
+    if marked_assingments:
+        if mark_file:
+            with open(marked_file_path, 'w') as f:
+                json.dump(marked_assignments, f)
+        else:
+            json.dump(marked_assignments, sys.stdout)
```

Comment:
My comment at amti/clis/review.py#L31-L35 suggests handling the default to STDOUT in the script's argument parsing. With that change, we'd no longer need a conditional here to check if the file is non-null.

Additionally, writing the marks out in [JSON Lines](http://jsonlines.org/) rather than JSON will make it easier to process the marks with line-oriented command-line tools like `grep`. It also has the benefit that we no longer need to check there are marked assignments:

Putting it all together, we can make these lines:

```python
with open(marked_file_path, 'w') as marked_file:
    marked_file.write('\n'.join(
        json.dumps(marked_assignment)
        for marked_assignment in marked_assignments
    ))
```

Reply:
I was under the impression that stdout couldn't be opened with the open function? That's why I coded it as I did.

Reply:
Sorry, I meant to mention: click has a great utility for correctly handling opening STDOUT / STDIN: [`click.open_file`](https://click.palletsprojects.com/en/7.x/utils/#intelligent-file-opening). Using that here in place of python's `open` built-in will do the right thing.